packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -0,0 +1,600 @@
+
+  TextEditingController _controller;
+  TextEditingController get _effectiveController => widget.controller ?? _controller;
+
+  FocusNode _focusNode;
+  FocusNode get _effectiveFocusNode => widget.focusNode ?? (_focusNode ??= FocusNode());
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.controller == null) {
+      _controller = TextEditingController();
+      _controller.addListener(updateKeepAlive);
+    }
+  }
+
+  @override
+  void didUpdateWidget(CupertinoTextField oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.controller == null && oldWidget.controller != null) {
+      _controller = TextEditingController.fromValue(oldWidget.controller.value);
+      _controller.addListener(updateKeepAlive);
+    }
+    else if (widget.controller != null && oldWidget.controller == null)
+      _controller = null;
+    final bool isEnabled = widget.enabled ?? true;
+    final bool wasEnabled = oldWidget.enabled ?? true;
+    if (wasEnabled && !isEnabled) {
+      _effectiveFocusNode.unfocus();
+    }
+  }
+
+  @override
+  void dispose() {
+    _focusNode?.dispose();
+    _controller?.removeListener(updateKeepAlive);
+    super.dispose();
+  }
+
+  void _requestKeyboard() {
+    _editableTextKey.currentState?.requestKeyboard();
+  }
+
+  RenderEditable get _renderEditable => _editableTextKey.currentState.renderEditable;
+
+  void _handleTapDown(TapDownDetails details) {
+    _renderEditable.handleTapDown(details);
+  }
+
+  void _handleTap() {
+    _renderEditable.handleTap();
+    _requestKeyboard();
+  }
+
+  void _handleTapCancel() {
+  }
+
+  void _handleLongPress() {
+    _renderEditable.handleLongPress();
+  }
+
+  @override
+  bool get wantKeepAlive => _controller?.text?.isNotEmpty == true;
+
+  @override
+  void deactivate() {
+    super.deactivate();
+  }
+
+  bool _shouldAppearForModeAndText(OverlayVisibilityMode mode, bool hasText) {
+    switch (mode) {
+      case OverlayVisibilityMode.never:
+        return false;
+      case OverlayVisibilityMode.always:
+        return true;
+      case OverlayVisibilityMode.editing:
+        return hasText;
+      case OverlayVisibilityMode.notEditing:
+        return !hasText;
+    }
+    assert(false);
+    return null;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context); // See AutomaticKeepAliveClientMixin.
+    assert(debugCheckHasDirectionality(context));
+    final Brightness keyboardAppearance = widget.keyboardAppearance;
+    final TextEditingController controller = _effectiveController;
+    final FocusNode focusNode = _effectiveFocusNode;
+    final List<TextInputFormatter> formatters = widget.inputFormatters ?? <TextInputFormatter>[];
+    final bool enabled = widget.enabled ?? true;
+    if (widget.maxLength != null && widget.maxLengthEnforced)
+      formatters.add(LengthLimitingTextInputFormatter(widget.maxLength));
+
+    Widget child = Padding(
```

Comment:
Given the proliferation of the term "child" in widget trees, can you name this variable based on what it represents within the CupertinoTextField UI?

Reply:
I'm breaking this build up a bit